crates/ruma-events/src/call/member.rs

```diff
@@ -449,44 +342,147 @@ mod tests {
         assert_eq!(
-            content.active_memberships(Some(one_second_ago)),
-            content.memberships.iter().collect::<Vec<&Membership>>()
+            content_legacy.active_memberships(Some(two_hours_ago)),
+            (vec![] as Vec<MembershipData<'_>>)
         );
+        // session do never expire
+        let content_session = create_call_member_event_content();
         assert_eq!(
-            content.active_memberships(Some(now)),
-            content.memberships.iter().collect::<Vec<&Membership>>()
+            content_session.active_memberships(Some(one_second_ago)),
+            content_session.memberships()
+        );
+        assert_eq!(content_session.active_memberships(Some(now)), content_session.memberships());
+        assert_eq!(
+            content_session.active_memberships(Some(two_hours_ago)),
+            content_session.memberships()
         );
-        assert_eq!(content.active_memberships(Some(two_hours_ago)), vec![] as Vec<&Membership>);
     }
 
     #[test]
     fn set_created_ts() {
-        let mut content_now = create_call_member_event_content();
-        let mut content_two_hours_ago = create_call_member_event_content();
-        let mut content_one_second_ago = create_call_member_event_content();
+        let mut content_now = create_call_member_legacy_event_content();
+        let mut content_two_hours_ago = create_call_member_legacy_event_content();
+        let mut content_one_second_ago = create_call_member_legacy_event_content();
         let (now, one_second_ago, two_hours_ago) = timestamps();
 
         content_now.set_created_ts_if_none(now);
         content_one_second_ago.set_created_ts_if_none(one_second_ago);
         content_two_hours_ago.set_created_ts_if_none(two_hours_ago);
+        assert_eq!(content_now.active_memberships(None), content_now.memberships());
+
         assert_eq!(
-            content_now.active_memberships(None),
-            content_now.memberships.iter().collect::<Vec<&Membership>>()
+            content_two_hours_ago.active_memberships(None),
+            vec![] as Vec<MembershipData<'_>>
         );
-
-        assert_eq!(content_two_hours_ago.active_memberships(None), vec![] as Vec<&Membership>);
         assert_eq!(
             content_one_second_ago.active_memberships(None),
-            content_one_second_ago.memberships.iter().collect::<Vec<&Membership>>()
+            content_one_second_ago.memberships()
         );
 
         // created_ts should not be overwritten.
         content_two_hours_ago.set_created_ts_if_none(one_second_ago);
         // There still should be no active membership.
-        assert_eq!(content_two_hours_ago.active_memberships(None), vec![] as Vec<&Membership>);
+        assert_eq!(
+            content_two_hours_ago.active_memberships(None),
+            vec![] as Vec<MembershipData<'_>>
+        );
+    }
+
+    #[cfg(feature = "unstable-msc4075")]
```

Comment:
This test and the following should be in the `notify` module. Actually, is there a reason this was moved out of the integration tests?

Reply:
Initially it all was part of one module. I moved out the notify and did not bring the tests with it.